src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp

```diff
@@ -3329,6 +3345,23 @@ CTranslatorRelcacheToDXL::IsIndexSupported(Relation index_rel)
 			GIN_AM_OID == index_rel->rd_rel->relam);
 }
 
+//---------------------------------------------------------------------------
+//	@function:
+//		CTranslatorRelcacheToDXL::IsIndexVisible
+//
+//	@doc:
+//		Check if index is visible in current transaction
+//
+//---------------------------------------------------------------------------
+BOOL
+CTranslatorRelcacheToDXL::IsIndexVisible(Relation index_rel)
+{
+	return !(index_rel->rd_index->indcheckxmin &&
+			 !TransactionIdPrecedes(
+				 HeapTupleHeaderGetXmin(index_rel->rd_indextuple->t_data),
+				 TransactionXmin));
```

Comment:
I have not found any other uses of either TransactionIdPrecedes nor HeapTupleHeaderGetXmin, nor TransactionXmin, nor TransactionIdEquals in the ORCA C++ code.

Reply:
As I understand it, ORCA does not take transactions into account at all when planning?

Reply:
Maybe then we need to fix execution, not planning?

Reply:
can we start using seqscan on executing when the planner has decided to use Index Scan? in addition, we will have inconsistency with the explain plan and the actual execution.

Reply:
Is it handled for postgres planner at planning stage? Give references, please

Reply:
https://github.com/arenadata/gpdb/blob/f4c4ee25754a85b824ce23087dc873b7c907e54d/src/backend/optimizer/util/plancat.c#L219-L231